### PR TITLE
Migrate caching things followup

### PR DESCRIPTION
**Pull request by taquitos**
Address comments from original PR #https://github.com/RevenueCat/purchases-ios/pull/658

### CODE REVIEW

**Comment by taquitos** [comment]
me rn:
Why are you like this

File: PurchasesCoreSwift/Caching/InMemoryCachedObject.swift [hunk 1]
```diff
@@ -6,9 +6,7 @@
 //  Copyright © 2021 Purchases. All rights reserved.
 //
 
-import Foundation
-
-open class InMemoryCachedObject<T> {
+public class InMemoryCachedObject<T> {
 
     private let accessQueue = DispatchQueue(label: "InMemoryCachedObjectQueue", attributes: .concurrent)
     private var lastUpdated: Date?
```
Comment (vegaro) [inline PurchasesCoreSwift/Caching/InMemoryCachedObject.swift:14]: There was a comment here https://github.com/RevenueCat/purchases-ios/pull/658#discussion_r674338482 that I believe is unaddressed

**Review by vegaro — APPROVED** [review]
I looked into the comments in the other PR and I think there's just one unaddressed. Other than that lgtm

File: PurchasesCoreSwift/Caching/InMemoryCachedObject.swift (hunk 1 above)
Comment (taquitos) [reply PurchasesCoreSwift/Caching/InMemoryCachedObject.swift:14]: Oh yeah! I'm used to doing `private` vars/lets at the top for some reason, hard habit to break.

Comment (taquitos) [reply PurchasesCoreSwift/Caching/InMemoryCachedObject.swift:14]: Fixed!

**Review by aboedo — APPROVED** [review]
looks good. the only thing I'd add is https://github.com/RevenueCat/purchases-ios/pull/658/files#r680190260, happy to move that conversation here

**Pull request merged**